aws-cpp-sdk-core/include/aws/core/endpoint/BuiltInParameters.h

```diff
@@ -0,0 +1,35 @@
+/**
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+
+#pragma once
+
+#include <aws/core/endpoint/EndpointParameter.h>
+#include <aws/core/client/ClientConfiguration.h>
+#include <aws/core/utils/memory/stl/AWSVector.h>
+
+namespace Aws
+{
+    namespace Endpoint
+    {
+        class AWS_CORE_API BuiltInParameters
+        {
+        public:
+            using EndpointParameter = Aws::Endpoint::EndpointParameter;
+
+            virtual ~BuiltInParameters(){};
+
+            const EndpointParameter& GetParameter(const Aws::String& name) const;
+            void SetParameter(EndpointParameter param);
+            void SetStringParameter(Aws::String name, Aws::String value);
```

Comment:
why not have name and value be const references?

Reply:
pass-by-value and move
also, it avoids potential issue of bad_alloc at the callee, because now caller is going to allocate memory for these strings.